### ISSUE TITLE
QA all workload summaries across Workload Mapper presets

### ISSUE THREAD

**Issue by jwsiejk** [jwsiejk/accountlist#381]
## Goal
QA and harden the AI Workload Discovery Mapper so **Workload Summary** and **BOM Summary** work consistently across every selectable workload, not just FSI — Fraud Detection & Investigation.

## Context
The FSI — Fraud Detection & Investigation workload is now close to the desired behavior:

- Workload Summary explains the workload in plain English for a sales counterpart.
- BOM Summary maps the workload to DDN-informed architecture/reference-pattern building blocks without creating an actual BOM.

Now validate and tune the same experience across all workloads.

## Workloads to test
Run the same QA flow against every workload:

### FSI
- Fraud Detection & Investigation
- Risk Modeling / Stress Testing
- Trading / Quant Research
- Compliance / Document Intelligence
- RAG Knowledge Assistant
- Model Training / Fine-tuning

### HPC / AI
- Large-scale AI Training
- Inference Platform
- Scientific Simulation
- Genomics / Life Sciences
- Media / Rendering Pipeline

### Custom
- Custom Workload using the generic example preset

## Manual QA flow for each workload
For each workload:

1. Select the workload.
2. Click `Auto-populate realistic example for this workload`.
3. Generate `Workload Summary`.
4. Generate `BOM Summary`.
5. Validate output quality using the criteria below.

## Workload Summary expected behavior
The Workload Summary should:

- Start with the workload category/name/purpose.
- Explain why the workload matters to the business.
- Explain what the customer is trying to accomplish.
- Explain what the technical environment needs to support.
- Use captured values naturally, not field names.
- Avoid quoting source fields back to the user.
- Avoid phrases like:
  - `I'd like to summarize`
  - `as stated by`
  - `which translates to`
  - `the input says`
  - `the field says`
- Mention missing information only when `missingInputs` actually contains missing fields.
- If no required discovery inputs are missing, say that no required discovery inputs are currently flagged as missing and point to BOM Summary for deeper architecture mapping.

## BOM Summary expected behavior
The BOM Summary should:

- Start with the required disclaimer:

```text
This is not a final BOM. It is a DDN-informed architecture mapping based on the current workload inputs.
```

- Explain the closest DDN-informed architecture/reference pattern.
- Use the workload-specific `ddnReferencePattern.buildingBlocks` as the primary source.
- Avoid generic headings like:
  - Compute
  - Storage / Data Platform
  - Network
  - AI Software
  - Security / Governance
  - Services
  when more specific DDN-informed building blocks exist.
- Avoid raw internal field labels like:
  - dailyIngestRange
  - filePattern
  - queryConcurrency
  - sizingInputs.haDrRequirements
- Use actual captured values naturally in the explanation.
- Read like a synthesized architecture/BOM-readiness narrative, not a rendered data object.
- Include validation-before-BOM guidance.
- End with the required closing note:

```text
This is an example starting point based on current DDN public reference architecture and platform themes. Actual sizing and final BOM require DDN/customer-specific validation, benchmark expectations, and constraints.
```

- Never generate:
  - actual BOM
  - exact SKUs
  - node counts
  - GPU counts
  - pricing
  - quote
  - final sizing
  - official one-to-one DDN BOM claim

### Risk Modeling / Stress Testing
Workload Summary should emphasize portfolio exposure, regulatory readiness, scenario impact, reproducibility, and compressing risk windows.

BOM Summary should map to:
- Governed historical risk data platform
- Scenario data pipeline
- Parallel read / analytics execution layer
- Scenario output and result aggregation layer
- Governance, lineage, and reproducibility controls
- Performance validation for batch/intraday run windows

### Trading / Quant Research
Workload Summary should explain that the business is trying to validate trading ideas and research signals faster.

BOM Summary should map to:
- Time-series / feature data active platform
- High-concurrency research access layer
- Metadata / indexing layer for feature and backtest artifacts
- Backtesting / analytics execution layer
- Low-latency data access validation
- Governance controls for proprietary strategy data

### Compliance / Document Intelligence
Workload Summary should emphasize reducing manual review effort, improving audit/regulatory response, citation-backed evidence, and explainability.

BOM Summary should map to:
- Governed document ingest pipeline
- Chunking / embedding pipeline
- Metadata indexing layer
- Vector / retrieval layer
- Explainability and citation/evidence layer
- Security, residency, audit, and access-control layer

### RAG Knowledge Assistant
Workload Summary should explain helping employees find trusted answers faster across governed internal knowledge sources.

BOM Summary should map to:
- Enterprise knowledge ingestion pipeline
- Metadata and indexing layer
- Embedding / vector retrieval layer
- Model serving / inference layer
- RBAC/ABAC governed access layer
- Retrieval freshness validation

### Model Training / Fine-tuning
Workload Summary should emphasize adapting models to domain language/processes, improving task quality, and maintaining governed reproducibility.

BOM Summary should map to:
- Curated fine-tuning dataset layer
- Training/validation execution layer
- Checkpoint and model registry layer
- High-throughput training data layer
- GPU training / fine-tuning compute layer
- Benchmark and sizing validation before BOM

### Large-scale AI Training
Workload Summary should emphasize building differentiated AI capabilities or domain models, keeping GPUs fed, and reducing model iteration time.

BOM Summary should map to:
- Large-scale corpus ingest and prep layer
- Distributed training data path layer
- Checkpoint and recovery layer
- GPU training cluster / AI factory compute layer
- High-speed interconnect / fabric
- GPU utilization and throughput benchmark plan

### Inference Platform
Workload Summary should emphasize serving AI capabilities reliably to users/apps with predictable latency and concurrency.

BOM Summary should map to:
- Model serving and routing layer
- Hot data and feature access layer
- Policy, safety, and observability layer
- Cache/session state layer
- Multi-tenant governance/security
- Latency/concurrency validation

### Scientific Simulation
Workload Summary should emphasize completing simulations faster and shortening time from run to insight.

BOM Summary should map to:
- Scheduler-integrated HPC compute fabric
- EXAScaler-style parallel file/data platform
- Checkpoint / output handling layer
- High-speed interconnect
- Post-processing / visualization workflow
- Job profile and checkpoint validation

### Genomics / Life Sciences
Workload Summary should emphasize improving sample/pipeline turnaround while preserving traceability, reproducibility, and regulated data handling.

BOM Summary should map to:
- Sequencing/pipeline ingest layer
- Metadata/indexing layer for many small files
- High-throughput pipeline data layer
- Analytics / AI interpretation layer
- Governance, retention, and reproducibility layer
- Regulated data access controls

### Media / Rendering Pipeline
Workload Summary should emphasize delivering creative/rendering work faster under deadline pressure.

BOM Summary should map to:
- High-throughput shared asset platform
- Metadata/indexing layer for media assets
- GPU render/inference pool
- Workflow orchestration / queue layer
- Production burst ingest and throughput validation
- Project-level access controls and archive policy

### Custom Workload
Custom should not fall apart.

The summaries should:
- use the custom workload name/category/description
- infer the business reason from process improved and success criteria
- choose a DDN-informed pattern based on selected AI pattern
- avoid saying important fields are missing if they are filled

## Implementation guidance
If any workload output is weak:

1. Improve the relevant preset data in `workload-mapper-data.ts`.
2. Improve `workload-mapper-ddn-reference.ts` so that workload has richer building blocks.
3. Tune `summarize-prompt.ts` or `bom-summary-prompt.ts` only if the weakness is prompt behavior rather than source data.

Do not overfit only to Fraud Detection.

## Suggested automated/internal QA helper
If useful, add a dev-only helper or test utility that confirms every preloaded workload has:

- example preset
- DDN reference pattern
- at least 3 DDN-informed building blocks
- validationBeforeBom entries
- sourceBasis entries

Do not expose unnecessary debug UI to users.

## Validation commands
Run:

```bash
npm run lint
npm run typecheck
npm test -- --runInBand
npm run build
rg -i "walkthrough|walkthrough mode|screen-share friendly discovery narrative" ui/partner-hub/components/workload-mapper ui/partner-hub/lib/workload-mapper ui/partner-hub/app/api/workload-mapper || true
rg -i "chip-?8|steam[- ]?trains" . || true
```

If commands are unavailable, document closest equivalent.

## Acceptance criteria
- Workload Summary works well for all preloaded workloads and custom workload.
- BOM Summary works well for all preloaded workloads and custom workload.
- No workload produces a generic category-only BOM Summary when specific DDN-informed building blocks exist.
- No workload summary quotes source fields or exposes internal field names.
- No BOM Summary exposes raw internal field names or schema labels.
- Every workload has a realistic example preset.
- Every workload has a DDN reference pattern with useful building blocks.
- No actual BOM/SKUs/node counts/GPU counts/pricing/final sizing is generated.
- No Walkthrough Mode references remain.
- No CHIP-8 / Steam Trains references are introduced.
